research/gaia/pegasus/pegasus/src/event/io/mod.rs

```diff
@@ -92,7 +92,10 @@ impl Encode for Events {
     fn write_to<W: WriteExt>(&self, writer: &mut W) -> ::std::io::Result<()> {
         match self {
             &Events::Batched(ref events) => events.write_to(writer),
-            _ => unimplemented!("[Unreachable]"),
+            &Events::Single(ref event) => {
```

Comment:
You do not need to write `writer.write_u32()` for the Batched case? Otherwise how can you handle read. In addition, it looks like write_u8 is enough.